lib/wisper/sidekiq.rb

```diff
@@ -1,12 +1,27 @@
+require 'yaml'
 require 'wisper'
 require 'sidekiq'
 
 require 'wisper/sidekiq/version'
 
 module Wisper
+
+  # based on Sidekiq 4.x #delay method, which is not enabled by default in Sidekiq 5.x
+  # https://github.com/mperham/sidekiq/blob/4.x/lib/sidekiq/extensions/generic_proxy.rb
+  # https://github.com/mperham/sidekiq/blob/4.x/lib/sidekiq/extensions/class_methods.rb
+
   class SidekiqBroadcaster
+    class Worker
+      include ::Sidekiq::Worker
+
+      def perform(yml)
+        (subscriber, event, args) = ::YAML.load(yml)
+        subscriber.public_send(event, *args)
+      end
+    end
+
     def broadcast(subscriber, publisher, event, args)
-      subscriber.delay.public_send(event, *args)
+      Worker.perform_async(::YAML.dump([subscriber, event, args]))
```

Comment:
Hello 👋 

This is nice, but again you loose all the control over the job as you cannot set `sidekiq_options` anywhere (as mentioned in #13).

Injecting `.set` call before `.perform_async` should allow for some more freedom without much of boiler plate. Maybe `subscriber` could declare optional class method of some sort (`def self.worker_options` or whatever) in order to pass those options as a hash to `set` per `subscriber` class, wdyt?

/cc @k-rudy (as that PR in wisper wasn't merged until today so I guess solution above tries to work around the problem :p)

Another possible solution is to discard whole wisper async-API and all the tooling and create a subscriber that just calls your-worker-of-choice with all the options you might need - which doesn't seems so bad all things considered - maybe wisper shouldn't try to support asynchronous support for all those platforms in the first place? 🤔 

Just my 2cents, that's for all the effort and work! ❤️

Reply:
I've always intended to support passing of options to the broadcaster in Wisper v3.0, but when this actually gets started I do not know.

I didn't know about the `#set` method :smile: 

Having an optional class method which can return configuration for Sidekiq sounds like it has potential. I know people have asked about being able to set things like the queue name, e.g. #15. 

As you say the other option is to have the async call inside a regular, non-async, Wisper listener.

Reply:
So maybe in favor of wrapping this up relying on that `set` + optional class method might be a good enough solution for moving things forward? Sidekiq 5 was released over a year ago (dang times goes fast!) so it would be cool if this gem support newest version somehow :)

If you need any help with that lemme know - I might find some spare time for this!

Reply:
Yes, I think your suggestion would work well.

This PR can be merged but I really need someone to try it out by pinning to this branch in Gemfile and make sure it works. I don't have the opportunity to do any testing at the moment. I am guessing most people have turned the [delay method back on](https://github.com/mperham/sidekiq/wiki/Delayed-extensions) when using Sidekiq 5 so the currently released version still works.

But I would prefer those who don't want to enable delay to still use this gem with Sidekiq 5.

Please feel free to start a PR to add the proposed feature. But maybe we need to merge this PR first so you can branch from master.

Reply:
To verify the usage would be something like:

```ruby
class MyListener
  def self.on_person_created(attributes)
    # send an email
  end

  def self.sidekiq_options
    { queue: "special" }
  end
end
```